AnnService/src/Core/BKT/BKTIndex.cpp

```diff
@@ -228,6 +240,7 @@ namespace SPTAG
             for (DimensionType i = 0; i <= checkPos; i++) { \
                 SizeType nn_index = node[i]; \
                 if (nn_index < 0) break; \
+                if (nn_index >= m_pSamples.R()) break; \
```

Comment:
This will not happen if build and refine will not insert VID larger than R. This check will be done thousands of times. We should remove it from critical path.